ivadomed/testing.py

```diff
@@ -179,36 +188,63 @@ def run_inference(test_loader, model, model_params, testing_params, ofolder, cud
                     # re-init pred_stack_lst
                     pred_tmp_lst, z_tmp_lst = [], []
 
-                # add new sample to pred_tmp_lst, of size n_label X h X w ...
-                pred_tmp_lst.append(preds_idx_arr)
+                    # add new sample to pred_tmp_lst, of size n_label X h X w ...
+                    pred_tmp_lst.append(preds_idx_arr)
 
-                # TODO: slice_index should be stored in gt_metadata as well
-                z_tmp_lst.append(int(batch['input_metadata'][smp_idx][0]['slice_index']))
-                fname_tmp = fname_ref
+                    # TODO: slice_index should be stored in gt_metadata as well
+                    z_tmp_lst.append(int(batch['input_metadata'][smp_idx][0]['slice_index']))
+                    fname_tmp = fname_ref
```

Comment:
I think this does not work: if `pred_tmp_lst` is not fed, then we never go into the "NEW COMPLETE VOLUME" section --> Niftis aren't saved.
I tried to undo this change locally and it seems to fix the issue.
What are your thoughts?

Reply:
Yes absolutely! While modifying the code I didn't indent these lines correctly and forgot to test for 2D models! Did the modifications :) everything should be good now

Reply:
I ve just got this when re running it:
```
  File "/home/GRAMES.POLYMTL.CA/chgroc/ivado-medical-imaging/ivadomed/metrics.py", line 71, in dice_score
    raise ValueError("Shape mismatch: im1 and im2 must have the same shape.")
ValueError: Shape mismatch: im1 and im2 must have the same shape.
```
But it may be because of me.. looking deeper at it.